### ISSUE TITLE
[feat] Secondary nav initial render improvement

### ISSUE THREAD

**Issue by zhawkins** [RedHat-UX/red-hat-design-system#1046]
### Description

We've got light DOM CSS that ships along with secondary nav (Yay!) but the "open by default on mobile" state we chose is a performance regression (Boo!).

<img width="1423" alt="image" src="https://github.com/RedHat-UX/red-hat-design-system/assets/2671663/bbfd7889-8b7e-4a2f-986b-3255074d563f">

https://www.webpagetest.org/video/compare.php?tests=230601_BiDcB8_9SQ,230504_BiDcC9_CX7

### Suggested solution

I believe the problem is there's not a great no-or-broken-js way to collapse the menu by default but allow it to be accessible. 

I've pulled together an example of how we might be able to accomplish this with the target selector: 

https://codepen.io/zackhawkins/full/BaqXwBz

@hellogreg threw together a better/different option with the following demo using a checkbox approach:

> This isn't perfect, but it works pretty well with mouse, keyboard, and/or screen reader. I built upon a Heydon Pickering version of the checkbox hack--which suits nav menus better than some other cases where I've seen it. Gave the checkbox a `button` role and used `aria-live` to announce visually-hidden expanded and collapsed text (as opposed to a JS-toggled `aria-expanded` attribute) to screen readers. Currently set up so the expanded menu list pushes down other content, though absolute positioning could make it overlap, if you want.

Demo: https://cdpn.io/pen/debug/BaqXwwR
Code: https://codepen.io/hellogreg/pen/BaqXwwR?editors=1100

**Comment by bennypowers** [comment]
is noscript a target here? do you have usage stats that we can reference to determine how much of our audience is noscript? can we solve this by defaulting to closed and adding dsd for fouc?

**Comment by zhawkins** [comment]
noscript is not a target. This is about initial render (having the nav in a collapsed state by default) and because it's in a collapsed state by default, providing a fallback to allow the user to open the nav if JavaScript fails to load or something breaks.

**Comment by zeroedin** [comment]
After discussion with @hellogreg we talked through current possible noscript fixes and each just present a different accessibility tradeoff to users.   The future goal would be DSD providing a `<details>` element as the closed mobile menu that would load closed and provide users access sans javascript.  Noted there are also still safari shortcomings around details current announce open/close state.

**Comment by hellogreg** [comment]
Yeah, `<details>` currently doesn't play well with Mobile Safari and VoiceOver (though is fine in Mac Safari). It announces neither a default role nor (as @zeroedin said) expanded/collapsed state. Putting an explicit "button" role on `<summary>` will announce the role in iOS, but it then also stops announcing expanded/collapsed state in _Mac_ Safari!

I know VoiceOver [used to announce state](https://www.scottohara.me/blog/2018/09/03/details-and-summary.html) with older versions of iOS (e.g., 11 and 12). But for some reason [does not announce state](https://www.scottohara.me/blog/2022/09/12/details-summary.html) with recent versions.

**Comment by zhawkins** [comment]
I'm not opposed to that, but mind you that Safari is a good chunk of our mobile traffic on RHDC. After all, this is the "everything has gone wrong" fallback. 

Currently all mobile users on slower connection speeds or lower powered devices (and a lot of our mobile traffic is Chrome) see the menu open by default and then shuffle close.
